Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Hole_filling/Triangulate_hole_polyline.h

```diff
@@ -1210,16 +1210,87 @@ class Triangulate_hole_polyline {
  ************************************************************************/
 // /!\ points.first == points.last
 
+template <typename Traits>
+bool is_planar_2(
+  const std::vector<typename Traits::Point_3>& points,
+  const typename Traits::Vector_3& avg_normal,
+  const typename Traits::FT max_distance,
+  const Traits& traits) {
+
+  typedef typename Traits::FT FT;
+  typedef typename Traits::Point_3 Point_3;
+  typedef typename Traits::Plane_3 Plane_3;
+  typedef typename Traits::Compute_squared_distance_3 Squared_distance_3;
+
+  const Squared_distance_3 squared_distance_3 =
+    traits.compute_squared_distance_3_object();
+
+  const std::size_t n = points.size() - 1; // the first equals to the last
+  if (n < 3)
+    return false; // cant be a plane!
+
+  // Compute centroid.
+  FT cx = FT(0), cy = FT(0), cz = FT(0);
+  for (std::size_t i = 0; i < n; ++i) {
+    const Point_3& p = points[i];
+    cx += p.x();
+    cy += p.y();
+    cz += p.z();
+  }
+  cx /= static_cast<FT>(n);
```

Comment:
Can we static cast from an int  to a FT?   Isn't a FT constructible from int? @sloriot 
Don't we have a centroid function?

Reply:
Actually, we do have the centroid function. You are right. I will change that. It is so simple that I forgot about it.

Reply:
The centroid function requires PCA, which we would like to avoid from the very beginning. So I have aborted this change.